Issues running autogen.sh under spack

When compiling flux under spack, autogen.sh fails.  @trws and I were able to fix it by adding an additional execution of aclocal after libtoolize.  @grondo suggested moving the discussion to an issue under flux-core.  He also suggested trying `autoreconf -i`.

Related spack issue: https://github.com/LLNL/spack/issues/1213
Related build log: https://gist.github.com/SteVwonder/9a5f2d7b839cb0ecafc9ed08abe7f1a7

Hmm, I still can't get flux-core to build under spack, looks like `libffi.so.6` is not in LD_LIBRARY_PATH?
```
Traceback (most recent call last):
  File "_core_build.py", line 3, in <module>
    ffi = FFI()
  File "/g/g0/grondo/git/spack/opt/spack/linux-redhat6-x86_64/gcc-4.4.7/py-cffi-1.1.2-3elvpt5xlzvhwdgizfd4s6d4t5dcg2nj/lib/python2.7/site-packages/cffi-1.1.2-py2.7-linux-x86_64.egg/cffi/api.py", line 56, in __init__
    import _cffi_backend as backend
ImportError: libffi.so.6: cannot open shared object file: No such file or directory
make[5]: *** [_core.c] Error 1

$ find ~/git/spack -name libffi.so.6 
/g/g0/grondo/git/spack/opt/spack/linux-redhat6-x86_64/gcc-4.4.7/libffi-3.2.1-zcwtp45bs4zbd22clwxgmuoypwu7kehs/lib64/libffi.so.6
```

Know what I'm doing wrong?

--- a/autogen.sh
+++ b/autogen.sh
@@ -1,16 +1,13 @@
 #!/bin/sh
-
-echo "Running aclocal ... "
-aclocal -I config
-echo "Running libtoolize ... "
+#
+# Run an extra libtoolize before autoreconf to ensure that
+# libtool macros can be found if libtool is in PATH, but its
+# macros are not in default aclocal search path.
+#
+echo "Running libtoolize --automake --copy ... "
 libtoolize --automake --copy 
-echo "Running autoheader ... "
-autoheader
-echo "Running automake ... "
-touch ChangeLog
-automake --copy --add-missing 
-echo "Running autoconf ... "
-autoconf
+echo "Running autoreconf --verbose --install -I config"
+autoreconf --verbose --install -I config
 echo "Cleaning up ..."
 mv aclocal.m4 config/
 rm -rf autom4te.cache

